File: panaderia-front/src/app/core/services/factura.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { FacturaRequest, FacturaResponse } from '../models/factura.model';

@Injectable({ providedIn: 'root' })
export class FacturaService {
  private apiUrl = `${environment.apiV1Url}/facturas`;

  constructor(private http: HttpClient) {}

  crear(factura: FacturaRequest): Observable<FacturaResponse> {
    return this.http.post<FacturaResponse>(this.apiUrl, factura);
  }

  listar(): Observable<FacturaResponse[]> {
    return this.http.get<FacturaResponse[]>(this.apiUrl);
  }

  buscarPorId(id: number): Observable<FacturaResponse> {
    return this.http.get<FacturaResponse>(`${this.apiUrl}/${id}`);
  }

  buscarPorNumero(numero: string): Observable<FacturaResponse> {
    return this.http.get<FacturaResponse>(`${this.apiUrl}/numero/${numero}`);
  }

  anular(id: number, motivo: string): Observable<FacturaResponse> {
    return this.http.patch<FacturaResponse>(`${this.apiUrl}/${id}/anular`, null, {
      params: new HttpParams().set('motivo', motivo)
    });
  }

  resumenDiario(): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/resumen-diario`);
  }
}

File: panaderia-api/src/main/java/com/panaderia/api/service/FacturaService.java
package com.panaderia.api.service;

import com.panaderia.api.dto.request.FacturaRequest;
import com.panaderia.api.dto.response.FacturaResponse;
import java.util.List;
import java.util.Map;

public interface FacturaService {
    FacturaResponse crear(FacturaRequest request, String usuarioCajero);
    FacturaResponse buscarPorId(Long id);
    FacturaResponse buscarPorNumero(String numero);
    List<FacturaResponse> listarTodas();
    FacturaResponse anular(Long id, String motivo);
    Map<String, Object> obtenerResumenDiario();
}
